### PR TITLE
Issue/1965 add ability to only show first letter of name fields

### PR DESCRIPTION
**Pull request by omarkasem**
- Fixes https://github.com/GravityKit/GravityView/issues/1965

💾 [Build file](https://www.dropbox.com/scl/fi/8mra6acrsxxr1fs0z8b91/gravityview-2.28.0-53dfa9143.zip?rlkey=9c1qb6de8jug610mibug1p8fj&dl=1) (53dfa9143).

### CODE REVIEW

File: includes/fields/class-gravityview-field-name.php [hunk 1]
```diff
@@ -22,8 +22,44 @@ class GravityView_Field_Name extends GravityView_Field {
...
+
+
+	public function get_content( $output = '', $entry = array(), $field_settings = array(), $field = array() ) {
+		/** Overridden by a template. */
+		if ( ! empty( $field['field_path'] ) ) {
+			return $output; 
+		}
+
+	
+		return 'test';
```
Comment (mrcasual) [inline includes/fields/class-gravityview-field-name.php:39]: @omarkasem, does this belong here?

File: templates/fields/field-name-html.php [hunk 2]
```diff
@@ -16,8 +16,25 @@
...
+	$initials = '';
+
+	foreach ( $names as $name ) {
+		$initials .= strtoupper( $name[0] ) . '.';
+	}
+
+	$display_value = trim($initials);
 }
+
+echo apply_filters('gk/gravityview/fields/name/initials', $display_value, $gravityview);
```
Comment (mrcasual) [inline templates/fields/field-name-html.php:40]: @omarkasem, is this a new filter? Please document.

File: includes/fields/class-gravityview-field-name.php (hunk 1 above)
Comment (omarkasem) [reply includes/fields/class-gravityview-field-name.php]: @mrcasual Yes my bad I forgot to remove that

**Comment by mrcasual** [comment]
@Mwalek, please test. Try using Arabic, Cyrillic and other scripts. Test the `gk/gravityview/fields/name/initials` filter as well.

**Comment by Mwalek** [comment]
@mrcasual  Overall, everything worked as expected. For Asian languages like Chinese, the entire first character was shown rather than a single initial, which is likely expected behavior due to how these scripts represent names.

The behavior was a bit weird when using emojis and parentheses, but not sure we want to support such edge cases (see image).

Screenshot 2024-09-18 at 3 39 20 PM

Some of my tests can be observed at the following URL:  
[https://basecamp.try.gravitykit.com/display-pull-2140-view/](https://basecamp.try.gravitykit.com/display-pull-2140-view/)

### Filter Testing

I also tested the filter `gk/gravityview/fields/name/initials` using the following code:

```php
add_filter('gk/gravityview/fields/name/initials', 'customize_initials', 10, 2);

function customize_initials($display_value, $gravityview) {
    
    $display_value = strtolower($display_value);

    return $display_value;
}
```
All good! The filter successfully altered the display of initials (English ones), converting them to lowercase as expected.
